README.md

```diff
@@ -1,4 +1,140 @@
-# MicroCluster
+# Microcluster
 ## Introduction
-MicroCluster is a helper package for bootstrapping and managing a dqlite cluster.
-Based on [go-dqlite](https://github.com/canonical/go-dqlite).
+
+Microcluster is a Go library that provides an easy to use framework for creating and managing highly available [dqlite](https://github.com/canonical/dqlite) clusters. It provides an extensible API and database which can be used to directly connect to your underlying services.
+
+Extend Microcluster with your own API endpoints, schema and API extensions, lifecycle actions, and additional network listeners.
+
+## Import Microcluster
+
+Get the latest LTS release of Microcluster:
+
+```
+$ go get github.com/canonical/microcluster/v2@latest
+```
+
+## Usage
+For a step-by-step practical example of creating a Microcluster project, check out the [example package](example). Examine the code or build and test it out with the included Makefile.
+
+All of Microcluster's state is stored in the `StateDir` directory. Define a new Microcluster app:
+
+```go
+m, err := microcluster.App(microcluster.Args{StateDir: "/path/to/state"})
+if err != nil {
+    // ...
+}
+```
+
+Extensions to Microcluster must be defined in `DaemonArgs`:
```

Comment:
"Extensions" feels really overloaded to me; `APIExtensions` and `ExtensionsSchema` are _really_ different things. Could we say something like "Define application services with `DaemonArgs`"?